### ISSUE TITLE
DicomElement validation varies with null values

### ISSUE THREAD

**Issue by esmadau** [fo-dicom/fo-dicom#1590]
**Describe the bug**
When validating on a DicomElement directly, it handles nulls gracefully. However, a NullReferenceException is thrown if using the VR validation off of the DicomElement.

**To Reproduce**
DicomElement de = dicomDataset.GetDicomItem<DicomElement>(DicomTag.AcquisitionDateTime);
de.Validate(); // handles null value fine
de.ValueRepresentation.ValidateString(nullValue); // will throw the exception

**Expected behavior**
Both eventually would call the same `de.ValueRepresentation.ValidateString(nullValue)` code, but de.Validate() [uses list comprehension and handles nulls gracefully](https://github.com/fo-dicom/fo-dicom/blob/5.0.3/FO-DICOM.Core/DicomElement.cs#L261). I would expect these two to behave the same, whichever is more correct by the DICOM standard.

**Screenshots or test DICOM files**
n/a

**Environment**

Fellow Oak DICOM version: 5.0.3.0
OS: Windows 11 Enterprise
Platform: .NET 7

**Comment by mrbean-bremen** [comment]
This is somewhat inconsistent: only the validators for AS, IS, LO, LT and PN do a null check - though this may be accidental, as `IsNullOrEmpty` seems to be used to check for empty strings, not null strings.  
One could probably argue that this is a documentation issue (e.g. the argument shall always be a single value of a possibly multi-valued tag, and should not be null), though in that case it should throw an `ArgumentException` if passed a null value.

**Comment by esmadau** [comment]
Coming here to add to this. There is another issue where 
creating and adding an element with a value and then getting the DicomElement and calling GetString on it works fine:
dicomDataset.Add(new DicomSignedLong(DicomTag.PregnancyStatus, new int[] { 1 }));

However, calling a DicomElement's GetString on this will throw the out of range exception:
dicomDataset.Add(new DicomSignedLong(DicomTag.PregnancyStatus, new int[] { }));

While not directly related to validation, it is related to DicomElement and how it is parsing values.

**Comment by gofal** [comment]
Here two different things are mixed up.

One thing is the validation of a ValueRepresentation. E.g. the DICOM standard very clearly defines how a value of VR DateTime or CodeString have to look like.

The other thing is then when validating a DicomTag. here there are more properties to validate, not only the VR but also the ValueMultiplicity for example.
That said, some DicomTags have a VM from exactly 1, there the validation of the DicomElement is directly the same as the validation of the corresponding ValueRepresentation.
Some other DicomTags have a VM from 1 to n, then the value is split up and the validation of the VR is applied to each of the values. But the validation should throw if you pass in null.
And some other DicomTags have VM from 0 to n, then null is a very valid value.

So your expected behavior "I would expect these two to behave the same, whichever is more correct by the DICOM standard." will not be implemented.

You should call the Validation method of a ValueReprenstaion only if you have the concrete single value that you want to validate. and `null` is not a valid value for all of the VRs.

But we could throw a DicomValidationException instead of a NullReferenceException.Would that be more consistent?

**Comment by mrbean-bremen** [comment]
> That said, some DicomTags have a VM from exactly 1, there the validation of the DicomElement is directly the same as the validation of the corresponding ValueRepresentation.

This is not completely true. A VM of 1 does not mean that the tag cannot be empty. This depends on the IOD, and a tag with a VM of 1 can be a type 2 or type 3 tag in some SOP classes, meaning that it can exist and be empty. As we do not do SOP class-specific validation, we can only check if the tag does not contain more than 1 value, and we generally should probably never throw on null values in the generic validation code. We could do it only for tags that are always mandatory (like SOPInstanceUID or SeriesInstanceUID), but this would be somewhere else in the application logic.

**Comment by gofal** [comment]
Agree. Thanks for clarify. The requirement of some DicomTags (if they are Type 3 Optional, Type 1 Required, or something between) depends on SOP classes.

But for the concrete example above with the AcquisitionDateTime Atribute.

    DicomElement de = dicomDataset.GetDicomItem(DicomTag.AcquisitionDateTime);
    de.Validate(); // handles null value fine
    de.ValueRepresentation.ValidateString(nullValue); // will throw the exception

When adding null value as Tag AcquisitonDateTime (which is of ValueMultiplicity 1 and Optional 3 for all of the SOPClasses that i checked), then the validation tries to split it up into several values and wants to check if more than 1 value was passed. The validation checked that no value was passed and therefore internally has an empty list of values. Therefore no further validation is done.
So the expected behavior is that this validation should succeed. And this is also the current behavior of fo-dicom.

On the other hand, when you take a null value and ask fo-dicom if null would be a valid DT value, then regarding to standard:
![image](https://github.com/user-attachments/assets/300b222f-cbc5-4798-9269-993c06863068)
the DT value has to contain at least the YYYY component, and therefore a null value has to be invalid.
So the expected behavior is that this validation should fail. This is what has to be changed in fo-dicom, there should not be a NullReferenceException, but a ValidationException.

Do we agree on that?
Or would one rather expect, that empty values should pass the Validation, too and a ValidationException is only raised if there is a value that does not match the definition in DICOM standard?

**Comment by mrbean-bremen** [comment]
> Or would one rather expect, that empty values should pass the Validation, too and a ValidationException is only raised if there is a value that does not match the definition in DICOM standard?

I would rather expect that, yes. As it is allowed to have no value (at least for some SOP classes), we should handle that as valid, and only check if the number of values exceeds the defined VM, and if each value conforms to the specs as given in PS 3.5.

**Comment by esmadau** [comment]
I think at the very least throwing a DicomValidationException instead of letting it go to NullReferenceException would be better as NullReferenceException is really very unexpected from something that is operating in production and it sounds like here it is really a validation issue

**Comment by mrbean-bremen** [comment]
Reading over this again, I take back what I wrote before about excepting empty values. @gofal is right that there is a difference between validating the tag values (which will be done for each value separately, and always pass for empty values), and the validation of a single value, which cannot be null. I agree with @esmadau that passing a null value to `ValueRepresentation.ValidateString` should throw a `DicomValidationException` (while it currently either throws  `NullReferenceException`, or just passes.

I'm not sure about the use case here, as usually you want to do the validation on the tag values, while the validation of each single value is more of an internal implementation. But as ``ValueRepresentation.ValidateString` is public and can be passed null, it shall at least behave consistently.
